File: pascal/classes/classes.pas
{$MODE OBJFPC}

program Classes;

type MoneyBox = class
  public
    name: string;
    saved_amount: double;
    constructor create();
    procedure save(amount: double);
end;

constructor MoneyBox.create();
begin
  name := '';
  saved_amount := 0;
end;

procedure MoneyBox.save(amount: double);
begin
  saved_amount := saved_amount + amount;
end;

var
  piggy: MoneyBox;

begin
  piggy := MoneyBox.create();

  piggy.name := 'Miss Piggy';

  writeln('Money box name: ', piggy.name);
  writeln('Money box saved amount: ', piggy.saved_amount:0:2);

  piggy.save(20);

  writeln('Money box saved amount: ', piggy.saved_amount:0:2);

  piggy.save(100);

  writeln('Money box saved amount: ', piggy.saved_amount:0:2);
end.

File: ruby/classes/classes.rb
#!/usr/bin/ruby2.7

class MoneyBox
  attr_writer :name
  attr_reader :name, :saved_amount
  def save(amount)
    @saved_amount = @saved_amount ? (@saved_amount + amount) : amount
  end
end

piggy = MoneyBox.new()
piggy.name = "Miss Piggy"
puts("Money box name: #{piggy.name}")
puts("Money box saved amount: #{piggy.saved_amount}")

piggy.save(20)

puts("Money box saved amount: #{piggy.saved_amount}")

piggy.save(100)

puts("Money box saved amount: #{piggy.saved_amount}")
